legend-pure-core/legend-pure-m4/src/main/java/org/finos/legend/pure/m4/coreinstance/primitive/date/DateFormat.java

```diff
@@ -56,18 +61,14 @@ public static <T extends Appendable> T format(T appendable, String formatString,
                             timeZoneId.append(next);
                             escaped = false;
                         }
-                        else if (inQuotes)
+                        else if (next == '"')
                         {
-                            timeZoneId.append(next);
+                            inQuotes = !inQuotes;
                         }
                         else if (next == ']')
```

Comment:
Since you're eliminating the inQuotes condition, I think this condition needs an extra conjunct so that we're not ending while we're still in quotes.
```suggestion
                        else if ((next == ']') && !inQuotes)
```